test/archive.cpp

```diff
@@ -199,99 +202,121 @@ class CapturedStderr
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.invalid_mimelistpos.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.invalid_mimelistpos.zim",
     "mimelistPos must be 80.\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.invalid_checksumpos.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.invalid_checksumpos.zim",
     "Checksum position is not valid\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.outofbounds_first_direntptr.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.outofbounds_first_direntptr.zim",
     "Invalid dirent pointer\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.outofbounds_last_direntptr.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.outofbounds_last_direntptr.zim",
     "Invalid dirent pointer\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.outofbounds_first_title_entry.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.outofbounds_first_title_entry.zim",
     "Invalid title index entry.\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.outofbounds_last_title_entry.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.outofbounds_last_title_entry.zim",
     "Invalid title index entry.\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.outofbounds_first_clusterptr.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.outofbounds_first_clusterptr.zim",
     "Invalid cluster pointer\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.nonsorted_dirent_table.zim",
-    "Dirent table is not properly sorted:\n"
-    "  #0: A/main.html\n"
-    "  #1: -/favicon\n"
-  );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.nonsorted_title_index.zim",
+  for(auto& testfile: getDataFilePath("invalid.nonsorted_dirent_table.zim")) {
+    std::string expected;
+    if (testfile.category == "normal") {
+      expected = "Dirent table is not properly sorted:\n"
+                 "  #0: A/main.html\n"
+                 "  #1: -/favicon\n";
+    } else {
+      expected = "Dirent table is not properly sorted:\n"
+                 "  #0: C/main.html\n"
+                 "  #1: C/favicon.png\n";
+    }
+    EXPECT_BROKEN_ZIMFILE(testfile.path, expected)
+  }
+
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.nonsorted_title_index.zim",
     "Title index is not properly sorted.\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.bad_mimetype_list.zim",
+  TEST_BROKEN_ZIM_NAME(
+    "invalid.bad_mimetype_list.zim",
     "Error getting mimelists.\n"
   );
 
-  EXPECT_BROKEN_ZIMFILE(
-    "./data/invalid.bad_mimetype_in_dirent.zim",
-    "Entry M/Language has invalid MIME-type value 1234.\n"
-  );
+  for(auto& testfile: getDataFilePath("invalid.bad_mimetype_in_dirent.zim")) {
+    std::string expected;
+    if (testfile.category == "normal") {
+      expected = "Entry M/Language has invalid MIME-type value 1234.\n";
+    } else {
+      expected = "Entry M/Scraper has invalid MIME-type value 1234.\n";
+    }
+    EXPECT_BROKEN_ZIMFILE(testfile.path, expected)
+  }
 }
```

Comment:
Partly agree.
It would be better on the test side yes.
But then we would have a evolution in the script generating the test data without a easy way to know which version to use.

We should handle this issue correctly when we will design a correct testing suite.